File: app/src/main/java/com/pixels/blockies/app/game/figures/FigureO.java
package com.pixels.blockies.app.game.figures;

public class FigureO extends AbstractFigure {

    private int[][][] figures = new int[][][]{
            {
                    {3, 3},
                    {3, 3},
            }
    };

    @Override
    protected int getFigureCount() {
        return figures.length;
    }

    @Override
    public int[][] get() {
        return figures[getCurrentRotation()];
    }

    @Override
    public int[][] getNext() {
        return figures[getNextRotation()];
    }
}

File: app/src/main/java/com/pixels/blockies/game/game/figures/FigureO.kt
package com.pixels.blockies.game.game.figures

class FigureO : AbstractFigure() {
    private val figures = arrayOf(arrayOf<IntArray?>(intArrayOf(3, 3), intArrayOf(3, 3)))
    override fun getFigureCount(): Int {
        return figures.size
    }

    override fun get(): Array<IntArray?> {
        return figures[currentRotation]
    }

    override fun getNext(): Array<IntArray?> {
        return figures[nextRotation]
    }
}